webapp/graphite/storage.py

```diff
@@ -14,58 +13,96 @@
 from graphite.logger import log
 from graphite.node import LeafNode
 from graphite.intervals import Interval, IntervalSet
-from graphite.finders.utils import FindQuery
-from graphite.readers.utils import MultiReader
-from graphite.worker_pool.pool import get_pool, pool_apply
+from graphite.finders.utils import FindQuery, BaseFinder
+from graphite.readers import MultiReader
+from graphite.worker_pool.pool import get_pool, pool_exec, Job, PoolTimeoutError
 from graphite.tags.utils import get_tagdb
 
 
-def get_finder(finder_path):
+def get_finders(finder_path):
     module_name, class_name = finder_path.rsplit('.', 1)
     module = import_module(module_name)
-    return getattr(module, class_name)()
+    cls = getattr(module, class_name)
+
+    if getattr(cls, 'factory', None):
+        return cls.factory()
+
+    # monkey patch so legacy finders will work
+    finder = cls()
+    finder.fetch = BaseFinder.fetch
+    finder.find_multi = BaseFinder.find_multi
+
+    return [finder]
 
```

Comment:
This change is what allows a finder module to create multiple finders, and is responsible for adding `fetch` and `find_multi` to legacy finders that don't extend `BaseFinder`.